packages/runtime/container-runtime/src/summarizerTypes.ts

```diff
@@ -0,0 +1,139 @@
+import { Deferred } from "@fluidframework/common-utils";
+import {
+    IFluidRouter,
+    IFluidRunnable,
+    IFluidLoadable,
+} from "@fluidframework/core-interfaces";
+import { ContainerWarning, IDeltaManager } from "@fluidframework/container-definitions";
+import {
+    IDocumentMessage,
+    ISequencedDocumentMessage,
+} from "@fluidframework/protocol-definitions";
+import { ISummaryStats } from "@fluidframework/runtime-definitions";
+import { IConnectableRuntime } from "./runWhileConnectedCoordinator";
+
+declare module "@fluidframework/core-interfaces" {
+    // eslint-disable-next-line @typescript-eslint/no-empty-interface
+    export interface IFluidObject extends Readonly<Partial<IProvideSummarizer>> { }
+}
+
+export const ISummarizer: keyof IProvideSummarizer = "ISummarizer";
+
+export interface IProvideSummarizer {
+    readonly ISummarizer: ISummarizer;
+}
+
+export interface ISummarizerInternalsProvider {
+    /** Encapsulates the work to walk the internals of the running container to generate a summary */
+    generateSummary(options: IGenerateSummaryOptions): Promise<GenerateSummaryData>;
+
+    /** Callback whenever a new SummaryAck is received, to update internal tracking state */
+    refreshLatestSummaryAck(
+        proposalHandle: string,
+        ackHandle: string,
+        summaryLogger: ITelemetryLogger,
+    ): Promise<void>;
+}
+
+export interface ISummarizingWarning extends ContainerWarning {
+    readonly errorType: "summarizingError";
+    readonly logged: boolean;
+}
+
+export interface ISummarizerRuntime extends IConnectableRuntime {
+    readonly logger: ITelemetryLogger;
+    readonly deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>;
+    readonly summarizerClientId: string | undefined;
+    nextSummarizerD?: Deferred<ISummarizer>;
+    closeFn(): void;
+    on(event: "batchEnd", listener: (error: any, op: ISequencedDocumentMessage) => void): this;
+    on(event: "disconnected", listener: () => void): this;
+    removeListener(event: "batchEnd", listener: (error: any, op: ISequencedDocumentMessage) => void): this;
+}
+
+export interface IGenerateSummaryOptions {
+    /** True to generate the full tree with no handle reuse optimizations; defaults to false */
+    fullTree?: boolean,
+    /** True to ask the server what the latest summary is first */
+    refreshLatestAck: boolean,
+    /** Logger to use for correlated summary events */
+    summaryLogger: ITelemetryLogger,
+}
+
+export interface IGeneratedSummaryStats extends ISummaryStats {
+    dataStoreCount: number;
+    summarizedDataStoreCount: number;
+}
+export interface IBaseSummaryData {
+    readonly referenceSequenceNumber: number;
+}
+export interface IGenerateSummaryData {
+    readonly summaryStats: IGeneratedSummaryStats;
+    readonly generateDuration: number;
+}
+export interface IUploadSummaryData {
+    readonly handle: string;
+    readonly uploadDuration: number;
+}
+export interface ISubmitSummaryData {
```

Comment:
I originally did it that way and then opted out since it felt like the types would become useless outside of this scope. However, I think for simplicity of reading I will revert it. Check the latest changes.